share/availability/discovery/discovery.go

```diff
@@ -235,3 +356,27 @@ func (d *Discovery) Advertise(ctx context.Context) {
 func (d *Discovery) Peers(ctx context.Context) ([]peer.ID, error) {
 	return d.set.Peers(ctx)
 }
+
+func (p Parameters) withDefaults() Parameters {
+	def := DefaultParameters()
+	if p.PeersLimit == 0 {
+		p.PeersLimit = def.PeersLimit
+	}
```

Comment:
We should not override it. 0 is an allowed value and needed to disable the limit